src/main/kotlin/lotto/model/Rank.kt

```diff
@@ -0,0 +1,20 @@
+package lotto.model
+
+import lotto.model.seller.Money
+
+enum class Rank(val prize: Money, val matchingCount: Int) {
+    First(Money(2_000_000_000), 6),
+    Second(Money(30_000_000), 5),
+    Third(Money(1_500_000), 5),
+    Fourth(Money(50_000), 4),
+    Fifth(Money(5_000), 3);
+
+    companion object {
+        fun of(matchingCount: Int, isMatchedBonus: Boolean): Rank? =
+            if (matchingCount == Second.matchingCount) {
+                if (isMatchedBonus) Second else Third
+            } else {
+                Rank.entries.find { rank -> rank.matchingCount == matchingCount }
+            }
```

Comment:
`이중 if 문`보다 `when` 사용하시는건 어떻게 생각하세요? 조금 더 간결해질 수 있을 것 같아요!

Reply:
아.. 다시 생각해보니까 굳이 이중 조건문을 사용할 필요는 없을 것 같네요

```suggestion
            if (matchingCount == Second.matchingCount && isMatchedBonus) {
                Second
            } else {
                Rank.entries.lastOrNull { rank -> rank.matchingCount == matchingCount }
            }
```